classes/migration/upgrade/v3_5_0/I9707_WeblateUILocales.php

```diff
@@ -15,18 +15,206 @@
 namespace PKP\migration\upgrade\v3_5_0;
 
 use Illuminate\Database\PostgresConnection;
+use Illuminate\Database\Query\Builder;
 use Illuminate\Support\Facades\DB;
 use PKP\install\DowngradeNotSupportedException;
 use PKP\migration\Migration;
 
-class I9707_WeblateUILocales extends Migration
+abstract class I9707_WeblateUILocales extends Migration
 {
+    protected string $CONTEXT_TABLE = '';
+    protected string $CONTEXT_SETTINGS_TABLE = '';
+    protected string $CONTEXT_COLUMN = '';
+
     /**
      * Run the migrations.
      */
     public function up(): void
     {
-        $localesTable = [
+        $affectedLocales = $this->getAffectedLocales();
+
+        // update all locale and primary_locale columns
+        $schemaLocName = (DB::connection() instanceof PostgresConnection) ? 'TABLE_CATALOG' : 'TABLE_SCHEMA';
+        $renameLocale = fn (string $l) => collect(DB::select("SELECT DISTINCT TABLE_NAME FROM INFORMATION_SCHEMA.COLUMNS WHERE COLUMN_NAME = ? AND {$schemaLocName} = ?", [$l, DB::connection()->getDatabaseName()]))
+            ->each(function (\stdClass $sc) use ($l, $affectedLocales) {
+                foreach ($affectedLocales as $uiLocale => $weblateLocale) {
+                    DB::table($sc->TABLE_NAME ?? $sc->table_name)->where($l, '=', $uiLocale)->update([$l => $weblateLocale]);
+                }
```

Comment:
This loop can be turned into a single query:
`...->whereIn($l, $uiLocales)->update([$l => DB::raw("CASE {$l} WHEN X THEN A WHEN Y THEN B END")]);`.